casper/src/test/scala/coop/rchain/casper/helper/TestNode.scala

```diff
@@ -341,44 +317,16 @@ case class TestNode[F[_]: Timer](
   def syncWith(node1: TestNode[F], node2: TestNode[F], rest: TestNode[F]*): F[Unit] =
     syncWith(IndexedSeq(node1, node2) ++ rest)
 
-  def contains(blockHash: BlockHash) = casperEff.contains(blockHash)
+  def contains(hash: BlockHash) = MultiParentCasperImpl.blockReceived(hash)
+
   def knowsAbout(blockHash: BlockHash) =
     (contains(blockHash), RequestedBlocks.contains[F](blockHash)).mapN(_ || _)
 
   def shutoff() = transportLayerEff.clear(local)
 
-  def visualizeDag(startBlockNumber: Int): F[String] =
-    for {
-      ref <- Ref[F].of(new StringBuffer(""))
-      ser = new StringSerializer(ref)
-      result <- BlockAPI.visualizeDag[F, String](
-                 Int.MaxValue,
-                 apiMaxBlocksLimit,
-                 startBlockNumber,
-                 (ts, lfb) =>
-                   GraphzGenerator.dagAsCluster[F](
-                     ts,
-                     lfb,
-                     GraphConfig(showJustificationLines = true),
-                     ser
-                   ),
-                 ref.get.map(_.toString)
-               )
-    } yield result.right.get
-
-  /**
-    * Prints a uri on stdout that, when clicked, visualizes the current dag state using ttps://dreampuf.github.io.
-    * The dag's shape is passed as Graphviz's dot source code, encoded in the URI's hash.
-    */
-  def printVisualizeDagUrl(startBlockNumber: Int): F[Unit] =
-    for {
-      dot <- visualizeDag(startBlockNumber)
-      // Java's URLEncode encodes ' ' as '+' instead of '%20', see https://stackoverflow.com/a/5330239/52142
-      urlEncoded = URLEncoder.encode(dot, "UTF-8").replaceAll("\\+", "%20")
-      _ <- Sync[F].delay {
-            println(s"DAG @ $name: https://dreampuf.github.io/GraphvizOnline/#" + urlEncoded)
-          }
-    } yield ()
```

Comment:
This is not used.

Reply:
Not sure wdy mean by this?

Reply:
It's not used for testing. We can add tests for GraphViz separately.
I didn't want to spend time to fix necessary dependencies to create block API instance here.